loader/loader_test.go

```diff
@@ -238,6 +240,83 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		})
 	})
 
+	When("calling GetPreviousRelease", func() {
+		var newerRelease, mostRecentRelease *v1alpha1.Release
+
+		AfterEach(func() {
+			k8sClient.Delete(ctx, newerRelease)
+			k8sClient.Delete(ctx, mostRecentRelease)
+
+			// Wait until the releases are gone
+			Eventually(func() bool {
+				releases := &v1alpha1.ReleaseList{}
+				err := k8sClient.List(ctx, releases,
+					client.InNamespace(release.Namespace),
+					client.MatchingFields{"spec.releasePlan": release.Spec.ReleasePlan})
+				return err == nil && len(releases.Items) == 1
+			}).Should(BeTrue())
+		})
+
+		It("returns a NotFound error if no previous release is found", func() {
+			returnedObject, err := loader.GetPreviousRelease(ctx, k8sClient, release)
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(BeTrue())
+			Expect(returnedObject).To(BeNil())
+		})
+
+		It("returns the previous release if found", func() {
+			// We need a new release with a more recent creation timestamp
+			time.Sleep(1 * time.Second)
+
+			newerRelease = release.DeepCopy()
+			newerRelease.Name = "newer-release"
+			newerRelease.ResourceVersion = ""
+			Expect(k8sClient.Create(ctx, newerRelease)).To(Succeed())
+
+			// Wait until the new release is cached
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: newerRelease.Name, Namespace: newerRelease.Namespace}, newerRelease)
+			}).Should(Succeed())
+
+			returnedObject, err := loader.GetPreviousRelease(ctx, k8sClient, newerRelease)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(returnedObject).ToNot(BeNil())
+			Expect(returnedObject.Name).To(Equal(release.Name))
+		})
+
```

Comment:
should we also test the scenario I described on top? Two releases created at the same second? Or we are assuming that this in the real scenario should not happen?

Reply:
It should not happen and there's not much we can do there.

Reply:
Indeed. while testing there we noticed that if two releases (or more) are created in the same second we can not be sure which one is newer as the createdTimestamp precision is in seconds. In fact, nothing we can do about that.